client/App.jsx

```diff
@@ -26,44 +36,116 @@ const Texts = styled.div`
+    this.overviewClicked = React.createRef();
+    this.twitterClicked = React.createRef();
+    //this.handleScrollToElement = this.handleScrollToElement.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      overviewClicked: true,
+      photosClicked: false,
+      menuClicked: false,
+      reviewsClicked: false,
+      twitterClicked: false,
+      restaurant: '',
+    }
+  }
+
+  handleClick(event) {
+  this.setState({
+    overviewClicked: false,
+    photosClicked: false,
+    menuClicked: false,
+    reviewsClicked: false,
+    twitterClicked: false
+  })
+  var target = event.target.id;
+  this.setState({
+    [target]: !this.state[target]
+  })
+  this[target].current.scrollIntoView({ behavior: 'smooth', block: 'start' })
+  }
+
+  componentDidMount() {
+    this.getRestaurantByObjectId();
   }
+
+  getRestaurantByObjectId() {
+    $.ajax({
+      // hardcoded 17
+      url: '/overview/17',
+      method: 'GET',
+      contentType: 'application/json',
+      success: (data) => {
+        this.setState({
+          restaurant: data[0],
+        });
+      },
+    });
+  }
+
   render() {
     return(
       <div>
-        <div>
-          <Border>
-            <Texts>
-              Hello
+          <Border ref={this.overviewClicked}>
+            <Texts onClick={this.handleClick} id="overviewClicked" style={{color: (this.state.overviewClicked ? "#da3743" : null), borderBottom: (this.state.overviewClicked ? "2px solid #da3743": null) }}>
+              Overview
             </Texts>
-            <Texts>
-              Hello
+            <Texts onClick={this.handleClick} id="photosClicked" style={{color: (this.state.photosClicked ? "#da3743" : null),  borderBottom: (this.state.photosClicked ? "2px solid #da3743": null )}}>
+              Photos
             </Texts>
-            <Texts>
-              Hello
+            <Texts onClick={this.handleClick} id="menuClicked" style={{color: (this.state.menuClicked ? "#da3743" : null),  borderBottom: (this.state.menuClicked ? "2px solid #da3743": null )}}>
+              Menu
             </Texts>
-            <Texts>
-              Hello
+            <Texts onClick={this.handleClick} id="reviewsClicked" style={{color: (this.state.reviewsClicked ? "#da3743" : null),  borderBottom: (this.state.reviewsClicked ? "2px solid #da3743": null )}}>
+              Reviews
             </Texts>
-            <Texts>
-              Hello
+            <Texts onClick={this.handleClick} id="twitterClicked" style={{color: (this.state.twitterClicked ? "#da3743" : null),  borderBottom: (this.state.twitterClicked ? "2px solid #da3743": null)}}>
+              Twitter
             </Texts>
           </Border>
-        </div>
+        <ParentBorder>
+          <Title restaurant={this.state.restaurant}/>
+          <Stars restaurant={this.state.restaurant}/>
+        </ParentBorder>
+        {/* CODE BELOW SHOULD BE INGORED -- ONLY USED FOR SCROLL TESTING */}
```

Comment:
Should this still be here?

Reply:
No, this was for testing the automatic scroll on click